File: module-1-llm-apis/solutions/python/02_sdk_basic.py
"""Module 1.2: SDK multi-turn chatbot."""
from dotenv import load_dotenv
import anthropic

load_dotenv()

client = anthropic.Anthropic()
messages = []


def main():
    print("Multi-turn chatbot (type 'quit' to exit)\n")

    while True:
        user_input = input("You: ")
        if user_input.lower() == "quit":
            break

        messages.append({"role": "user", "content": user_input})

        response = client.messages.create(
            model="claude-sonnet-4-20250514",
            max_tokens=1024,
            messages=messages,
        )

        block = response.content[0]
        if block.type == "text":
            print(f"\nClaude: {block.text}\n")
            messages.append({"role": "assistant", "content": block.text})

        print(f"[{len(messages)} messages, {response.usage.input_tokens}+{response.usage.output_tokens} tokens]")


if __name__ == "__main__":
    main()

File: module-1-llm-apis/solutions/02-sdk-basic.ts
/**
 * Module 1.2: SDK multi-turn chatbot
 * Conversation = ever-growing messages array.
 */
import "dotenv/config";
import Anthropic from "@anthropic-ai/sdk";
import * as readline from "readline";

const client = new Anthropic();
const messages: Anthropic.MessageParam[] = [];

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = (q: string) => new Promise<string>((res) => rl.question(q, res));

async function main() {
  console.log("Multi-turn chatbot (type 'quit' to exit)\n");

  while (true) {
    const input = await ask("You: ");
    if (input.toLowerCase() === "quit") break;

    messages.push({ role: "user", content: input });

    const response = await client.messages.create({
      model: "claude-sonnet-4-20250514",
      max_tokens: 1024,
      messages,
    });

    const text = response.content[0];
    if (text.type === "text") {
      console.log(`\nClaude: ${text.text}\n`);
      messages.push({ role: "assistant", content: text.text });
    }

    console.log(`[${messages.length} messages, ${response.usage.input_tokens}+${response.usage.output_tokens} tokens]`);
  }

  rl.close();
}

main().catch(console.error);
